source/framework/src/BookStoreManager.cc

```diff
@@ -0,0 +1,238 @@
+
+// -- marlin headers
+#include <marlin/BookStoreManager.h>
+#include <marlin/Application.h>
+#include <marlin/Exceptions.h>
+#include <marlin/Processor.h>
+
+#include <marlin/book/StoreWriter.h>  
+
+#include <string>
+
+namespace marlin {
+
+#define INSTANCIATIONS_HIST(type) \
+  template book::Handle<book::Entry<type>> BookStoreManager::getObject<type>(\
+    const book::EntryKey&) const ;\
+                            \
+  template book::Handle<book::Entry<type>> BookStoreManager::bookHist<type>(\
+      const std::filesystem::path&,\
+      const std::string_view&,\
+      const std::string_view&,\
+      const std::array<\
+        const AxisConfig<typename type::Precision_t>*,\
+        type::Dimension>&,\
+      const BookFlag_t&) 
+
+  INSTANCIATIONS_HIST(Hist1F);
+  INSTANCIATIONS_HIST(Hist1D);
+  INSTANCIATIONS_HIST(Hist1I);
+  INSTANCIATIONS_HIST(Hist2F);
+  INSTANCIATIONS_HIST(Hist2D);
+  INSTANCIATIONS_HIST(Hist2I);
+  INSTANCIATIONS_HIST(Hist3F);
+  INSTANCIATIONS_HIST(Hist3D);
+  INSTANCIATIONS_HIST(Hist3I);
+
+  //--------------------------------------------------------------------------
+    
+  
+  void BookStoreManager::writeToDisk() const {
+    if (_storeFile == "") {
+      if (_entriesToWrite.empty())   {
+        _logger->log<MESSAGE>() << "No Output file set!\n";
+      } else {
+        _logger->log<WARNING>() << "No Output file set, but " 
+          << _entriesToWrite.size() << " entries to write!\n";
+      }
+    } else {
+      book::StoreWriter writer ( _storeFile ) ;
+      _bookStore.storeList(
+        writer,
+        _entriesToWrite.begin(), 
+        _entriesToWrite.end());
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  
+
+  void BookStoreManager::init( const Application *app ) {
+    if( isInitialized() ) {
+      MARLIN_THROW( "BookStoreManager::init: Already initialized !" ) ;
+    }
+    _application = app ;
+    _logger = _application->createLogger( "BookStoreManager" ) ;
+
+    std::shared_ptr<StringParameters> paras =  app->bookStoreParameters();
+    if (!paras) {
+      _logger->log<WARNING>() 
+        << "no <store> node exist on top level!\n"
+        << "\tNo Output file set!\n"
+        << "\tUse Default flags for Booking!\n";
+      _defaultFlag = BookFlags::MultiShared | BookFlags::Store ;  
+    } else {
+      _storeFile =
+        paras->getValue<std::string>(ParameterNames::OutPutFile, "");
+      if (_storeFile == "") {
+        try { 
+          _storeFile = std::filesystem::temp_directory_path();
+        } catch (const std::filesystem::filesystem_error&) {
+          _storeFile = "";
+        }
+        std::stringstream strs{};
+        strs << "MarlinMT_" << std::this_thread::get_id() << ".root";
+        _storeFile /= strs.str(); 
+        _logger->log<WARNING>()
+          << "no output file for store defined, will write in temporary file: "
```

Comment:
It is not a temporary file, just the name will be generated

Reply:
no MarlinMT_[pid].root in execution directory